### ISSUE TITLE
Gatttool hangs

### ISSUE THREAD

**Issue by Phhere** [ianrenton/playbulb-tools#1]
Hello,
thanks for sharing your code. But i have the problem that gatttool can not send any commands in non interactive mode. Did you changed some settings on your system? In interactive mode everything works.

But i found some more commands ;) 
https://github.com/Phhere/Playbulb/blob/master/Protocol.md

**Comment by ianrenton** [comment]
Thanks for finding more commands! Do you know if you've found commands for all the Playbulb features yet, or are there some you haven't been able to find? If some are still missing I'll try playing around some more and see if I can find other commands too.

I haven't changed any settings on my system to make `gattool` non-interactive mode work, but it's an undocumented feature so it may only work on a few versions. My version of `bluez` (which provides `gattool`) is 4.101-0ubuntu13.1. What version of `bluez` are you using? If yours is older (or maybe even newer) that could explain the difference.

My computer is unusual though: I'm using Ubuntu 14.04 armhf in a chroot on a Samsung Chromebook, so my kernel Bluetooth drivers may be from Google or Samsung rather than Ubuntu's open-source ones. So if there's no difference in our `bluez` versions it could be something to do with that. I'll try it on a 'real' computer at some point.

**Comment by Phhere** [comment]
I have only playbulb candles here so the commands are only tested with these. But it responds so nearly everythin between 0x001 and 0x002d as a handle. 
I am running archlinux so my bluez version is quiet new 5.28 on a thinkpad t440s.
The non interactive mode works with `gattool characteristics` and `gatttool char-read` commands but not with char-write so maybe there is something broken

**Comment by derchrisuk** [comment]
Hi,

thanks for looking into this.
As Phil mentioned, I also had the latest blue 5.28 installed, and could not get it to write in non-interactive.
So I downgraded again to 4.101-0ubuntu20, and can also write.
Looks like a bug introduced in one of the later versions?
Anyway, we can now continue to look at the protocol and have all handles documented.
Thanks

**Comment by derchrisuk** [comment]
As a side note, this might not be documented, but available from the CLI of gatttool

````gatttool --help-all````

**Comment by ianrenton** [comment]
@derchrisuk You're right, I hadn't noticed that there was a `--help-all` which gave the arguments for non-interactive mode.

I've updated the README to reflect our finding that BlueZ 4.101 seems OK but 5.28 doesn't work.

**Comment by Riz0** [comment]
@Phhere Were you ever able to get char-write working on arch in non-interactive mode? I tried downgrading to  4.101 as @ianrenton suggested to no avail. Char-write only seems to works in interactive mode.

**Comment by DigiH** [comment]
@Phhere @ianrenton @derchrisuk Hi all, did any of you get gatttool non-interactive mode with any version of BlueZ after 5.28> I'm currently on 5.50 and gatttool non-interactive mode still seems dead in the water. Any help or info would be greatly appreciated.
